vision_agent_tools/tools/florence2_sam2.py

```diff
@@ -0,0 +1,144 @@
+            if return_mask:
+                with torch.autocast(device_type="cuda", dtype=torch.bfloat16):
+                    masks, _, _ = self.image_predictor.predict(
+                        point_coords=None,
+                        point_labels=None,
+                        box=bboxes,
+                        multimask_output=False,
+                    )
+            for i in range(len(bboxes)):
+                objs[annotation_id] = ImageBboxAndMaskLabel(
+                    bounding_box=bboxes[i],
+                    mask=(
+                        masks[i, 0, :, :] if len(masks.shape) == 4 else masks[i, :, :]
+                    )
+                    if return_mask
+                    else None,
+                    label=prompt,
+                )
+                annotation_id += 1
+        return objs
+
+    @torch.inference_mode()
+    def handle_image(
+        self, image: Image.Image, prompts: list[str]
+    ) -> dict[int, dict[int, ImageBboxAndMaskLabel]]:
+        self.image_predictor.reset_predictor()
+        objs = self.get_bbox_and_mask(image.convert("RGB"), prompts)
+        return {0: objs}
+
+    @torch.inference_mode()
+    def handle_video(
+        self, video: VideoNumpy, prompts: list[str]
+    ) -> dict[int, dict[int, MaskLabel]]:
+        self.image_predictor.reset_predictor()
+        objs = self.get_bbox_and_mask(
+            Image.fromarray(video[0]).convert("RGB"), prompts, return_mask=False
+        )
+        with torch.autocast(device_type="cuda", dtype=torch.bfloat16):
+            inference_state = self.video_predictor.init_state(video=video)
+            for annotation_id in objs:
+                _, _, out_mask_logits = self.video_predictor.add_new_points_or_box(
+                    inference_state=inference_state,
+                    frame_idx=0,
+                    obj_id=annotation_id,
+                    box=objs[annotation_id].bounding_box,
+                )
+
+            annotation_id_to_label = {}
+            for annotation_id in objs:
+                annotation_id_to_label[annotation_id] = objs[annotation_id].label
+
+            video_segments = {}
+            for (
+                out_frame_idx,
+                out_obj_ids,
+                out_mask_logits,
+            ) in self.video_predictor.propagate_in_video(inference_state):
+                video_segments[out_frame_idx] = {
+                    out_obj_id: MaskLabel(
+                        mask=(out_mask_logits[i][0] > 0.0).cpu().numpy(),
+                        label=annotation_id_to_label[out_obj_id],
+                    )
+                    for i, out_obj_id in enumerate(out_obj_ids)
+                }
+            self.video_predictor.reset_state(inference_state)
+            return video_segments
+
+    @validate_call(config={"arbitrary_types_allowed": True})
+    @torch.inference_mode()
+    def __call__(
+        self, media: Image.Image | VideoNumpy, prompts: list[str]
```

Comment:
@dillonalaird @camiloaz should we handle the input value as either image or video as separate values as we do [here](https://github.com/landing-ai/vision-agent-tools/blob/main/vision_agent_tools/tools/internlm_xcomposer2.py#L77-L78)?

Reply:
yeah, agree. will do that.

Reply:
done